nixos/modules/virtualisation/qcow.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.virtualisation.qcow;
+in
+{
+  options = {
+    virtualisation.qcow = {
+      baseImageSize = mkOption {
+        type = with types; either (enum [ "auto" ]) ints.positive;
+        default = "auto";
+        example = 8192;
+        description = ''
+          The base size of the QCOW2 image in MiB.
+        '';
+      };
+      additionalImageSpace = mkOption {
+        type = types.int;
+        default = 512;
+        example = 1024;
+        description = ''
+          Additional disk space in MiB to be added to the image if <literal>virtualisation.qcow.baseImageSize</literal> is set to <literal>auto</literal>.
```

Comment:
Where does this space end up? I would guess in the root filesystem _partition_; is that correct?
It seems likely that filesystem metadata overhead eats away a bit of the space, so it seems best to be specific about what's what.